src/components/views/rooms/SendMessageComposer.tsx

```diff
@@ -682,6 +683,38 @@ export class SendMessageComposer extends React.Component<ISendMessageComposerPro
             return true; // to skip internal onPaste handler
         }
 
+        // Safari `Insert from iPhone or iPad`
+        // data.getData("text/html") returns a string like: <img src="blob:https://...">
+        if (data.types.includes("text/html")) {
+            const imgElementStr=data.getData("text/html");
+            const parser = new DOMParser();
+            const imgDoc=parser.parseFromString(imgElementStr,"text/html");
+
+            const imgSrc = imgDoc.querySelector("img").src;
+
+            fetch(imgSrc).then((response) => {
+                response.blob().then((imgBlob) => {
+                    const type = imgBlob.type;
+                    const safetype = getBlobSafeMimeType(type);
+                    const ext=type.split("/")[1];
+                    const file = new File([imgBlob],"safari_inserted_content."+ext,{type:safetype});
+                    ContentMessages.sharedInstance().sendContentToRoom(
+                        file,
+                        this.props.room.roomId,
+                        this.props.relation,
+                        this.props.mxClient,
+                        null,
```

Comment:
This likely isn't right, as a reply may be active

Reply:
Solved.

Reply:
Good point. Done!